src/ReflectionExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2015, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\ParserReflection;
+
+use ReflectionExtension as BaseReflectionExtension;
+
+/**
+ * Returns AST-based reflections from extensions.
+ */
+class ReflectionExtension extends BaseReflectionExtension implements IReflection
+{
+    /**
+     * @var null|ReflectionClass[] ParsedReflection wrapped classes.
+     */
+    private $classes;
+
+    /**
+     * @var null|ReflectionFunction[] ParsedReflection wrapped functions.
+     */
+    private $functions;
+
+    /**
+     * Has extension been loaded by PHP.
+     *
+     * @return true
+     *     Enabled extensions are always loaded.
+     */
+    public function wasIncluded()
+    {
+        return true;
+    }
+
+    /**
+     * Returns list of reflection classes
+     *
+     * @return array|\ReflectionClass[]
+     */
+    public function getClasses()
+    {
+        if (!isset($this->classes)) {
+            $classRefs = parent::getClasses();
+            $this->classes = [];
+            foreach ($classRefs as $origKey => $eachClass) {
+                $this->classes[$origKey] = new ReflectionClass($eachClass->name);
+            }
+        }
+
+        return $this->classes;
+    }
+
+    /**
+     * Returns list of reflection functions
+     *
+     * @return array|\ReflectionFunction[]
```

Comment:
1. Please remove `array|`, because `[]` ending of `\ReflectionFunction[]` already indicates to IDE that it's an array.
2. Shouldn't `\ReflectionFunction[]` be `ReflectionFunction[]` (namespaced version) of class instead?

Reply:
Again, I'll try to fix this without Scrutinizer throwing a tantrum. 👍